platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/BuildscriptResolutionIntegrationTest.groovy

```diff
@@ -0,0 +1,832 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.test.fixtures.dsl.GradleDsl
+
+/**
+ * Tests edge cases of buildscript configuration resolution.
+ *
+ * <p>Tests should cover cases of initscript, settings, standalone, and project buildscript configurations.</p>
+ */
+class BuildscriptResolutionIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
🤔 I'd consider breaking this class into 4, one for each type os script.

Reply:
I think ATM I would prefer to keep these in one place. If somebody adds tests that verify some behavior for one kind script, I'd like them to also verify them for the other types of scripts as well. 

Having them all in the same place means the tests for all 4 types of scripts are discoverable, but having them in separate places could make it easy to write a test for one script but not the others. 

I would consider an abstract class based solution, but at the moment I dont want to develop that infrastructure

Reply:
An abstract class solution seems like a good fit, but yeah, not a big concern.